protostar/commands/test/test_result_formatter.py

```diff
@@ -75,248 +66,53 @@ def _format_passed_test_case_result(
-    )
-
-    for key, value in failed_test_case_result.exception.execution_info.items():
-        info_items.append(f"{key}={log_color_provider.bold(value)}")
-
-    if len(info_items) > 0:
-        info = ", ".join(info_items)
-        first_line_items.append(log_color_provider.colorize("GRAY", f"({info})"))
-
-    result.append(" ".join(first_line_items))
-
-    result.append("\n")
-    result.append(str(failed_test_case_result.exception))
-    result.append("\n")
-
-    for metadata in failed_test_case_result.exception.metadata:
-        result.append(_get_formatted_metadata(metadata))
-        result.append("\n")
-
-    result.extend(_get_formatted_stdout(failed_test_case_result.captured_stdout))
-
-    return "".join(result)
+) -> FailedTestCaseResultMessage:
+    return FailedTestCaseResultMessage(failed_test_case_result=failed_test_case_result)
 
 
 def _format_broken_test_case_result(
     broken_test_case_result: BrokenTestCaseResult,
-) -> str:
-    result: List[str] = []
-    first_line_items: List[str] = []
-
-    first_line_items.append(f"[{log_color_provider.colorize('RED', 'BROKEN')}]")
-    formatted_file_path = _get_formatted_file_path(broken_test_case_result.file_path)
-    first_line_items.append(
-        f"{formatted_file_path} {broken_test_case_result.test_case_name}"
-    )
-
-    info_items = []
-
-    info_items.append(
-        _get_formatted_execution_time(broken_test_case_result.execution_time)
-    )
-
-    for key, value in broken_test_case_result.exception.execution_info.items():
-        info_items.append(f"{key}={log_color_provider.bold(value)}")
-
-    if len(info_items) > 0:
-        info = ", ".join(info_items)
-        first_line_items.append(log_color_provider.colorize("GRAY", f"({info})"))
-
-    result.append(" ".join(first_line_items))
-
-    result.append("\n")
-    result.append(str(broken_test_case_result.exception))
-    result.append("\n")
-
-    for metadata in broken_test_case_result.exception.metadata:
-        result.append(_get_formatted_metadata(metadata))
-        result.append("\n")
-
-    result.extend(_get_formatted_stdout(broken_test_case_result.captured_stdout))
-
-    return "".join(result)
+) -> BrokenTestCaseResultMessage:
+    return BrokenTestCaseResultMessage(broken_test_case_result=broken_test_case_result)
 
 
 def _format_passed_fuzz_test_case_result(
     passed_fuzz_test_case_result: PassedFuzzTestCaseResult,
-) -> str:
-    first_line_elements: List[str] = []
-    first_line_elements.append(f"[{log_color_provider.colorize('GREEN', 'PASS')}]")
-    formatted_file_path = _get_formatted_file_path(
-        passed_fuzz_test_case_result.file_path
-    )
-    first_line_elements.append(
-        f"{formatted_file_path} {passed_fuzz_test_case_result.test_case_name}"
+) -> PassedFuzzTestCaseResultMessage:
+    return PassedFuzzTestCaseResultMessage(
+        passed_fuzz_test_case_result=passed_fuzz_test_case_result
```

Comment:
These functions could be removed in favor of creating messages directly in `format_test_result`.